ci: add test data and GitHub Actions workflow

AngryMaciek opened the pull request:
* add small test input to run the pipeline on and compare output against (integration test); two RNA-Seq samples from [this dataset](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE69656), 5mln reads sampled from each fastq file, mapped against only chr21.
* add GitHub Actions CI mechanism: local run with both _conda_ and _singularity_ technologies as well as DAG and rulegraph runs
* add shields to README
* add GitHub Issues templates

ninsch3000 commented on install/environment.root.yml:
```diff
@@ -0,0 +1,32 @@
+###############################################################################
```
This file should be mentioned as an alternative mode of installation in the section `https://github.com/zavolanlab/PAQR2/blob/9b1f6a38cbde0be33ced2d45b1c35b3b311864af/README.md#3-dependencies-installation` of the README.

AngryMaciek replied:
OK!

ninsch3000 reviewed (changes requested):
Thanks a lot for including the CI @AngryMaciek , that's awesome.   
Some general comments:
- is it really necessary to compare *every* intermediate output file of PAQR (and consequently, have a reference of every intermediate file inside the repo), or could these checks be limited to some crucial files (And accordingly some of the intermediate test files be removed from the repo?   
- Could you include a section in the README for a user to test their installation of PAQR? Currently, the integration test is only set up as CI, but maybe a user would want to check whether their installation worked as well.
- Similarly, please add a paragraph to the "dependencies & installation" section, where you describe the "evironment.root.yaml" and how it could be used instead of installing Singularity manually before building the conda env from the plain "environment.yaml".

AngryMaciek commented:
> * is it really necessary to compare _every_ intermediate output file of PAQR (and consequently, have a reference of every intermediate file inside the repo), or could these checks be limited to some crucial files (And accordingly some of the intermediate test files be removed from the repo?

Yeah, I think it is a good practice. We do not need to test every function in each script obviously but I see no reason why would we need to remove the current comparisons from the CI. This is a comp. pipeline so in my opinion having a test (and expected output) for every step (rule) might be highly beneficial. Especially for us - developers, when we ever come back to this repository and decide to change some part of the workflow we could immediately see how the effect of that given change propagates. Also, imagine a situation where you change sth "irrelevant" at the very first step of the pipeline and the end results are unexpectedly different. Tracing back such problems is a pain but with such a checkpoint after every step we have more control and easier insight overall.
All this is good for the CI checks, for the installation check (below) the user would just run the pipeline on the test data without the `cmp` commands.

> * Could you include a section in the README for a user to test their installation of PAQR? Currently, the integration test is only set up as CI, but maybe a user would want to check whether their installation worked as well.

OK, done, I added _Testing the execution_ subsection and run it myself. It runs on the same data as in the CI but without these file checks.

> * Similarly, please add a paragraph to the "dependencies & installation" section, where you describe the "evironment.root.yaml" and how it could be used instead of installing Singularity manually before building the conda env from the plain "environment.yaml".

OK, I added the same section as we have it in _ZARP_.

AngryMaciek commented:
Included comments from @ninsch3000 :

- README update
- echo message at the end of the test

I have checked zarp repo and I think that including comparisons in the installation test as in zarp would be too much for me right now...

ninsch3000 commented on README.md:
```diff
@@ -1,3 +1,7 @@
+[![ci](https://github.com/zavolanlab/tandem-pas/workflows/CI/badge.svg?branch=main)](https://github.com/zavolanlab/tandem-pas/actions?query=workflow%3ACI)
```
This badge shows "no status". Is this desired behaviour?

ninsch3000 reviewed (approved):
Thanks a lot @AngryMaciek !   
In case you make another commit, could you add two spaces at end of  README line 122? Markdown doesn't display properly there...
